nixos/doc/rl-2211: document nsncd option (#197986)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2211.section.md

  for a transition period so that in time the ecosystem can switch without
  breaking compatibility with any supported NixOS release.

- `nscd` functionality, necessary to provide non-glibc-builtin NSS
  modules (such as `nss_systemd` or `nss_ldap`) can now be provided by

>   `nsncd`, by setting `services.nscd.enableNsncd` to `true`.

Reviewer: Maybe link nsncd's home page?

>   The glibc-provided `nscd` daemon wasn't very reliable, and it was not fully
>   possible to disable caching functionality, causing various issues and stuck
>   lookups.

Reviewer: ```suggestion
  The `nscd` daemon provided by glibc isn't very reliable, and it's not fully
  possible to disable caching functionality, causing various issues and failed
  lookups.
```

Author: Suggestion applied, thanks!

  The plan is to switch from `glibc-nscd` to `nsncd` by default for the release

>   after this.

Reviewer: These subpoints should be prefixed with `-` (I think).

Author: Makes sense. Updated.

>   The plan is to switch from `glibc-nscd` to `nsncd` by default for the release
>   after this.

Reviewer: ```suggestion
  We plan to use `nsncd` by default in NixOS 23.05.
```

Author: Suggestion applied, thanks!
